src/lib/sql/sql_translator.cpp

```diff
@@ -402,15 +467,63 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_join(const hsql::Join
                                                    : std::make_pair(*left_in_right_node, *right_in_left_node);
 
   // Joins currently only support one simple condition (i.e., not multiple conditions).
-  auto predicate_condition = translate_operator_type_to_predicate_condition(condition.opType);
+  auto predicate_condition = translate_operator_type_to_predicate_condition(join_condition->opType);
 
   auto join_node = std::make_shared<JoinNode>(join_mode, column_references, predicate_condition);
   join_node->set_left_child(left_node);
   join_node->set_right_child(right_node);
-
+  _include_join_conditions(join_node, LQPChildSide::Left, left);
+  _include_join_conditions(join_node, LQPChildSide::Right, right);
   return join_node;
 }
 
+void SQLTranslator::_include_join_conditions(const std::shared_ptr<AbstractLQPNode>& join_node, const LQPChildSide side,
+                                             const std::vector<const hsql::Expr*>& conditions) {
+  if (conditions.empty()) {
+    return;
+  }
+
+  std::vector<std::shared_ptr<AbstractLQPNode>> predicates;
+  for (auto condition : conditions) {
```

Comment:
I prefer `const auto*` but that's just me.

Reply:
Yes.